### PR TITLE
Reliability fix

### LINKED ISSUE #888: Reliability: Average interitem correlation 95%CI, do not include point estimate!
* JASP version:
0.13.1
* OS name and version:
Windows 10
* Analysis: 
Reliability
* Bug description:
In Reliability Statistics table, the point estimate for Average interitem correlation is .21 and the 95%CI are reported as .87 - .84
At the same time, in the Cronbach's alpha column 95%CI values are not displayed. I assume there are the two limits displayed wrong on the column Average interitem correlation
* Expected behaviour:
95%CI (if computable) should include the point estimate
Cronbach's alpha should have 95%CI

[data_Reliability.zip](https://github.com/jasp-stats/jasp-issues/files/4936836/data_Reliability.zip)

Best regards,
Marian

### PR DESCRIPTION
**Pull request by juliuspfadt**
fixes https://github.com/jasp-stats/jasp-issues/issues/888
This is a major bug, cause the interval columns of two estimators are switched. 
I tested it with two datasets. Maybe a reviewer could also try this again. 
In the future it might be good to fix the unit tests for reliability, so this does not have to be done manually.

### CODE REVIEW

**Comment by JorisGoosen** [comment]
@Kucharssim could you look at this?

And maybe add a unittest? You know how to do so right?

File: JASP-Engine/JASP/R/reliabilityAnalysisBayesian.R [hunk 1]
```diff
@@ -204,14 +204,17 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
       relyFit$Bayes$ifitem$samp$mean <- (matrix(NA_real_, ncol(dataset), 2))
       relyFit$Bayes$ifitem$samp$sd <- (matrix(NA_real_, ncol(dataset), 2))
       
-      ops <- .BayesianReliabilityDerivedOptions(options)
-      order <- ops[["order"]]
-      relyFit[["Bayes"]][["samp"]] <- relyFit[["Bayes"]][["samp"]][order]
-      relyFit[["Bayes"]][["chains"]] <- relyFit[["Bayes"]][["chains"]][order]
-      relyFit[["Bayes"]][["est"]] <- relyFit[["Bayes"]][["est"]][order]
+      # reorder for JASP
+      names_est <- names(relyFit$Bayes$est)
```
Comment (Kucharssim) [inline JASP-Engine/JASP/R/reliabilityAnalysisBayesian.R:208]:
```suggestion
      names_est <- names(relyFit[["Bayes"]][["est"]])
```

File: JASP-Engine/JASP/R/reliabilityAnalysisFrequentist.R [hunk 2]
```diff
@@ -197,11 +187,19 @@ reliabilityFrequentist <- function(jaspResults, dataset, options) {
         }
         relyFit$freq$ifitem$mean <- colMeans(dataset, na.rm = T)
         relyFit$freq$ifitem$sd <- apply(dataset, 2, sd, na.rm = T)  
+
+        # reorder for JASP
+        names_est <- names(relyFit$freq$est)
```
Comment (Kucharssim) [inline JASP-Engine/JASP/R/reliabilityAnalysisFrequentist.R:192]:
```suggestion
        names_est <- names(relyFit[["freq"]][["est"]])
```

**Review by Kucharssim — APPROVED** [review]
The bug is fixed, I left some minor suggestions (nice, but not necessary to address, it's just about R Style convention).

I am happy to help out with the unit test, but I cannot do so next week; only after that.

**Comment by JorisGoosen** [comment]
Ok, so I'll merge this now and then when @juliuspf comes back from vacation he could process the suggestions maybe.

For the unittests he also made https://github.com/jasp-stats/INTERNAL-jasp/issues/993 and I've assigned you both to it.
Maybe later @Kucharssim can help Julius out in how to make the unittests.

**Comment by juliuspfadt** [comment]
@Kucharssim Would you like the changes to list elements such as `relyFit$freq$est` into `relyFit[["freq"]][["est"]]` also for things such as `scaleTable$addColumnInfo` into `scaleTable[["addColumnInfo"]]` or `scaleTable$dependOn` into `scaleTable[["dependOn"]]`?

**Comment by JorisGoosen** [comment]
> 
> 
> @Kucharssim Would you like the changes to list elements such as `relyFit$freq$est` into `relyFit[["freq"]][["est"]]` also for things such as `scaleTable$addColumnInfo` into `scaleTable[["addColumnInfo"]]` or `scaleTable$dependOn` into `scaleTable[["dependOn"]]`?

Don't do that, it will break the table.
`addColumnInfo` and `dependOn` are functions, not list-entries.

**Comment by juliuspfadt** [comment]
Damn, so "replace all" was a mistake. thanks

**Pull request merged**